libs/ledger-live-common/src/bridge/generic-alpaca/prepareTransaction.ts

```diff
@@ -26,86 +22,57 @@ export function genericPrepareTransaction(
-        )
-      ).value;
+
+    const next: GenericTransaction = {
+      ...transaction,
+      assetOwner,
+      assetReference,
+    };
+
+    const useAllAmount = !!next.useAllAmount || ["stake", "unstake"].includes(next.mode ?? "");
+
+    // Anticipate use cases where token amounts impact fee calculuses
+    if (useAllAmount && next.subAccountId) {
+      const subAccount = account.subAccounts?.find(sub => sub.id === next.subAccountId);
+
+      next.amount = subAccount?.spendableBalance ?? new BigNumber(0);
     }
 
-    if (!bnEq(transaction.fees, new BigNumber(fees.toString()))) {
-      const next: GenericTransaction = {
-        ...transaction,
-        fees: new BigNumber(fees.toString()),
-        assetReference,
-        assetOwner,
-        customFees: {
-          parameters: {
-            fees: customParametersFees ? new BigNumber(customParametersFees.toString()) : undefined,
-          },
-        },
-      };
-
-      // propagate storageLimit fee parameter when present (ex: tezos)
-      const feeEstimation = await estimateFees(
-        transactionToIntent(
-          account,
-          {
-            ...transaction,
-          },
-          computeIntentType,
-        ),
-      );
-      const params = feeEstimation?.parameters;
-      if (params) {
-        const storageLimit = params["storageLimit"];
-        if (
-          storageLimit !== undefined &&
-          (typeof storageLimit === "bigint" ||
-            typeof storageLimit === "number" ||
-            typeof storageLimit === "string")
-        ) {
-          next.storageLimit = new BigNumber(storageLimit.toString());
-        }
-      }
+    const intent = transactionToIntent(account, next, computeIntentType);
+    const estimation = await estimateFees(intent);
+    const customFeesValue = next.customFees?.parameters?.fees; // e.g. Stellar
+
+    next.fees = customFeesValue ?? new BigNumber(estimation.value.toString());
+
+    const fieldsToPropagate = ["storageLimit"] as const;
+
+    for (const field of fieldsToPropagate) {
+      const parameter = estimation.parameters?.[field];
 
-      // align with stellar/xrp: when send max (or staking intents), reflect validated amount in UI
       if (
-        transaction.useAllAmount ||
-        transaction["mode"] === "stake" ||
-        transaction["mode"] === "unstake"
+        typeof parameter === "bigint" ||
+        typeof parameter === "number" ||
+        typeof parameter === "string"
       ) {
-        const { amount } = await validateIntent(
-          transactionToIntent(
-            account,
-            {
-              ...transaction,
-            },
-            computeIntentType,
-          ),
-        );
-        next.amount = new BigNumber(amount.toString());
+        next[field] = new BigNumber(parameter.toString());
       }
-      return next;
     }
 
-    return transaction;
+    // Fees are now fixed, compute max spendable native
+    if (useAllAmount && !next.subAccountId) {
```

Comment:
```suggestion
    if (useAllAmount && !next.subAccountId) {
      // Use estimation.parameters.amount if available
      if (estimation.parameters?.amount) {
        next.amount = new BigNumber(estimation.parameters.amount.toString());
      } else {
        next.amount = account.spendableBalance.gt(next.fees)
          ? account.spendableBalance.minus(next.fees)
          : new BigNumber(0);
      }
    }
```

Sadly Tezos might not be able to send the whole spendableBalance because of this historical issue: https://gitlab.com/tezos/tezos/-/issues/1754

Basically we have to leave a small dust margin value for Tezos, the "send max" isn't really "sending max" and always leave a few dust values on the balance (otherwise the tx will fail because of insufficient funds), the real max spendable value is propagated through `estimation.parameters.amount` in Tezos.